In [6]:
!pwd

/home/user/Tae_StudyCode/TransformersDataAugmentation/src/utils/datasets/trec


In [2]:
%cd TransformersDataAugmentation/src/utils/datasets/trec

/home/user/Tae_StudyCode/TransformersDataAugmentation/src/utils/datasets/trec


In [3]:
import pandas as pd

raw_dataset = pd.read_csv("train.tsv", delimiter='\t')
raw_dataset

Numeric                      How long is human gestation ?
0           Human  Who is known as `` the world 's oldest teenage...
1        Location           What is George Lucas 's e-mail address ?
2     Description         What are Cushman and Wakefield known for ?
3         Numeric  How many states did Richard Nixon carry in 1972 ?
4     Description    How does the tail affect the flight of a kite ?
...           ...                                                ...
4900     Location                      Where is the Keck telescope ?
4901     Location  What country did the Allies invade in World Wa...
4902  Description  What are those little blue reflectors in the m...
4903  Description  What would happen to Canada if Quebec separated ?
4904        Human           Who are the Challengers of the Unknown ?

[4905 rows x 2 columns]

In [8]:
''' exp_0_10 dataset '''
raw_dataset_0to10 = pd.read_csv("exp_0_10/train.tsv", delimiter='\t')
raw_dataset_0to10

numeric         what is columbia tristar 's phone number ?
0        numeric                how big is our galaxy in diameter ?
1        numeric  what is the most number of goals scored by a s...
2        numeric            how many varieties of apple are there ?
3        numeric             what year did montana become a state ?
4        numeric  how many lawyers are there in the state of new...
5        numeric    what chapter of the bible has the most verses ?
6        numeric              what is dick clark 's date of birth ?
7        numeric                               when is boxing day ?
8        numeric                        when is the thai new year ?
9          human                   who invented `` the muppets '' ?
10         human  what two commanders directed the forces in the...
11         human  who was the first elected mayor of washington ...
12         human  who played sally rogers on the dick van dyke s...
13         human  what apollo 11 astronaut minded the store whil...
14         human  who manufactures the software , `` photoshop '' ?
15         human  who was scrooge 's dead partner in dickens 's ...
16         human  who was the mother of the man who would not be...
17         human     who awarded the flying fickle finger of fate ?
18         human  what 's the middle name of movie producer jose...
19      location            where is one 's corpus callosum found ?
20      location   what caribbean island is northeast of trinidad ?
21      location             what city is found in the city of oz ?
22      location   what 's the fifth-largest country in the world ?
23      location  where does the weird shape of the dinner fish ...
24      location  what is the name of the planet that the ewoks ...
25      location  what asian gulf were the destroyers maddox and...
26      location  what u.s. city is known as the rubber capital ...
27      location  what country is the setting for edgar allan po...
28      location  what country boasts cawdor castle , glamis cas...
29   description                      what is thermal equilibrium ?
30   description  how did the hohenzollerns build their power ar...
31   description                         what happened to pompeii ?
32   description                             what is home banking ?
33   description                          what is an urban legend ?
34   description  what was al capone finally imprisoned for , in...
35   description                            how do birds have sex ?
36   description  what was the v-8 juice slogan : `` the tastebu...
37   description                 why are there olives in martinis ?
38   description  what is the history of spartacus , the roman s...
39        entity  what kind of patent did ermal fraze get in 1963 ?
40        entity       which ear can most people hear better with ?
41        entity  name the french historical period during the r...
42        entity           what determines the value of the pearl ?
43        entity         what language is mostly spoken in brazil ?
44        entity  what is the lens behind the iris in the eye ca...
45        entity                     what are halloween 's colors ?
46        entity     in beetle bailey , can you name sarge 's dog ?
47        entity  what fastener did whitcomb judson patent in 18...
48        entity  what game do steve mcqueen and edward g. robin...
49  abbreviation                 what is the full name of the plo ?
50  abbreviation                                      what is dsl ?
51  abbreviation                         what does nasa stand for ?
52  abbreviation                          what does ekg stand for ?
53  abbreviation                    what does `` b.y.o.b. '' mean ?
54  abbreviation  what is the abbreviated form of the national b...
55  abbreviation                          what does bud stand for ?
56  abbreviation  what is p.m. an abbreviation for , as in 5 p.m. ?
57  abbreviation  what is the abbreviation of the company name `.

### 각각의 bert로 augmentation한 결과
- cbert
- cmobert
- cmobertp

In [9]:
''' cbert 결과 '''
cbert_dataset_0to10 = pd.read_csv("exp_0_10/cbert/cbert_aug.tsv", delimiter='\t')
cbert_dataset_0to10

numeric        what is columbia tristar ' s phone number ?
0        numeric                how big is our galaxy in diameter ?
1        numeric  what is the most number of goals scored by a s...
2        numeric            how many varieties of apple are there ?
3        numeric             what year did montana become a state ?
4        numeric  how many kids are musicians in the state of ne...
5        numeric       what book of the bible has the most verses ?
6        numeric               what is dickburn ' s date of birth ?
7        numeric                               when is boxing day ?
8        numeric                        when is the thai new year ?
9          human             who invented ` ` the muppet show ' ' ?
10         human  what two commanders directed the forces in the...
11         human  when was the first elected mayor of washington...
12         human  who played sally rogers on the dick van dyke s...
13         human  what apollo 11 astronaut minded the store whil...
14         human  who manufactures the software for ` ` photosho...
15         human  who was scrooge ' s literary partner in dicken...
16         human  who was the mother of the man who actually not...
17         human     who awarded the flying fickle finger of fate ?
18         human  what ' s the middle name of movie producer jos...
19      location           where is one ' s corpus callosum found ?
20      location       what small island is northeast of trinidad ?
21      location             what city is found in the city of oz ?
22      location   what ' s the fifth - largest city in the world ?
23      location  where does the weird shape of the dinner fish ...
24      location  what is the name of the town that the ewoks li...
25      location  which asian gulf were the destroyers maddox an...
26      location  what u . s . city is known as the rubber capit...
27      location  what country is the setting for edgar allan po...
28      location  what country boasts cawdor castle , glampo cas...
29   description                             what is thermal rock ?
30   description  how did the hohenzollerns build their power ar...
31   description                         what happened to pompeii ?
32   description                                 what is home for ?
33   description                         what is an urban ecology ?
34   description   what was al caponeone imprisoned for , in 1931 ?
35   description                            how do birds have sex ?
36   description  what was the v - 8 juice slogan : ` ` the tast...
37   description                  why are there lime in martiniss ?
38   description  what is the history of spartacus , the trade s...
39        entity  what kind of patent did ermal coze hold in 1963 ?
40        entity       which ear can most people hear better with ?
41        entity  about the french historical period during the ...
42        entity           what determines the value of the pearl ?
43        entity         what language is mostly spoken in brazil ?
44        entity  what is the lens behind the iris in the eye ca...
45        entity                    what are halloween ' s colors ?
46        entity    in beetle bailey , can you name sarge ' s dog ?
47        entity  what sweetener did whitcomb judson patent in 1...
48        entity  what game do steve mcqueen and edward g . robi...
49  abbreviation                 what is the full name of the plo ?
50  abbreviation                                      what is dsl ?
51  abbreviation                         what does nasa stand for ?
52  abbreviation               what does ek philippines stand for ?
53  abbreviation           what does ` ' b . y . o . c . ' ' mean ?
54  abbreviation  what is the abbreviated form of the national b...
55  abbreviation                          what does bud stand for ?
56  abbreviation  what is o . m . an abbreviation for , as in 5 ...
57  abbreviation  what is the source of the company name ` gener.

In [10]:
''' c mod bert (= Bert Only MLM) 결과 '''
cmodbert_dataset_0to10 = pd.read_csv("exp_0_10/cmodbert/cmodbert_aug.tsv", delimiter='\t')
cmodbert_dataset_0to10

numeric        what is columbia tristar ' s phone number ?
0        numeric               what big is our galaxy in diameter ?
1        numeric  what is the statesman number of goals scored b...
2        numeric            how many varieties of apple are there ?
3        numeric             what year did stocked become a state ?
4        numeric  how many lawyers are there the the state of ne...
5        numeric  what chapter losses the bible has the most ver...
6        numeric             what is dick clark ' s date of birth ?
7        numeric                               what is boxing day ?
8        numeric                        what is the thai new year ?
9          human                 who invented ` ` the muppets ' ' ?
10         human  whatuted commanders directed the forceseyer th...
11         human  who was the first elected mayor of washington ...
12         human  who played sally rogers on the dick van stamme...
13         human  what apollo 11 astronaut mindedcked store whil...
14         human  who manufactures the software , ` ` photoshop ...
15         human  who was scrooge ' s dead partner in dickens ' ...
16         human  who was the mother of the man who would not be...
17         human     who awarded the flying fickle finger of fate ?
18         human  what ' s the middle name of movie producer jos...
19      location         where is one ' s corpus callos the found ?
20      location         what the island is northeast of trinidad ?
21      location            what city is laurel in the city of oz ?
22      location  what ' s the fifth - largest country in the wo...
23      location  where does the tong shape of the the fish knif...
24      location  what what the name of the planet that the ewok...
25      location  what asian gulf were the destroyers maddox and...
26      location  what u . s . city is known as theا capital of ...
27      location  what country is the setting for edgar allan po...
28      location  what country muller cawdor castle , glamis cas...
29   description                      what is thermal equilibrium ?
30   description  how did the ho 01zollerns build their power ar...
31   description                         what happened to pompeii ?
32   description                            pepsi is home banking ?
33   description                              what is an urban is ?
34   description  what marcus al capone finally imprisoned for ,...
35   description                           how do birds have from ?
36   description  what was the v - 8 juice slogan : ` ` the tast...
37   description                 why are there olive in martiniss ?
38   description  what average the history of spartacus , the ro...
39        entity  what kind of patent did ermal ʰze jewelry in 1...
40        entity       which ear can most people hear better with ?
41        entity  name the french in period during the reign of ...
42        entity       what determines the value of the plausible ?
43        entity         what language is mostly spoken in brazil ?
44        entity  what is the lens behind the iris in the eye ca...
45        entity                    what are halloween ' s colors ?
46        entity  in beetle wellington , can you name sarge ' s ...
47        entity    in fastener in whitcomb judson patent in 1893 ?
48        entity  what game do steve mcqueen and edward g . robi...
49  abbreviation               what is the fullurities of the plo ?
50  abbreviation                               what is ds stadium ?
51  abbreviation                           what does nasa stand ি ?
52  abbreviation                          what does ekg stand for ?
53  abbreviation            what does ` `f . y . o . b . ' ' mean ?
54  abbreviation  what is the abbreviatedper of the national bur...
55  abbreviation                          what does bud stand for ?
56  abbreviation  what is p . older . an abbreviation for , as i...
57  abbreviation  what is the abbreviation for the company name .

In [11]:
''' c mod bert p (= Bert MLM + NSM) 결과 '''
cmodbertp_dataset_0to10 = pd.read_csv("exp_0_10/cmodbertp/cmodbertp_aug.tsv", delimiter='\t')
cmodbertp_dataset_0to10

numeric        what is columbia tristar ' s phone number ?
0        numeric                how big is our galaxy in diameter ?
1        numeric  what is the average number of goals scored by ...
2        numeric            how many varieties of apple are there ?
3        numeric             what year did montana become a state ?
4        numeric  how many lawyers are there in the state of new...
5        numeric    what chapter in the bible has the most verses ?
6        numeric             what is dick clark ' s date of birth ?
7        numeric                               what is boxing day ?
8        numeric                        what is the thai new year ?
9          human                 who invented ` ` the muppets ' ' ?
10         human  what how commanders directed the forces during...
11         human  who was the first elected mayor of washington ...
12         human  who played sally rogers on the dick van dyke s...
13         human  what apollo 11 astronaut minded in store while...
14         human  who manufactures the software , ` ` photoshop ...
15         human  who was scrooge ' s dead partner in dickens ' ...
16         human  who was the mother of the man who would not be...
17         human   being awarded the flying fickle finger of fate ?
18         human  what ' s the middle name of movie producer jos...
19      location           where is one ' s corpus callosum found ?
20      location       what small island is northeast of trinidad ?
21      location           what city is located in the city of oz ?
22      location  what ' s the fifth - largest country in the wo...
23      location  where does the par shape of the medieval fish ...
24      location  what in the name of the planet that the ewoks ...
25      location  what asian gulf were the destroyers maddox and...
26      location  what u . s . city is known as the financial ca...
27      location  what country is the setting for edgar allan po...
28      location  what country is cawdor castle , glamis castle ...
29   description                      what is thermal equilibrium ?
30   description  how did the hohezollerns build their power aro...
31   description                         what happened to pompeii ?
32   description                             what is home banking ?
33   description                           what is an urban smirk ?
34   description  what was al capone finally imprisoned for , in...
35   description                          how do birds have wings ?
36   description  what was the v - 8 juice slogan : the ' the ta...
37   description                 why are there olive in martiniss ?
38   description  what is the history of spartacus , the roman s...
39        entity  what kind of patent did ermal arize obtain in ...
40        entity       which ear can most people hear better with ?
41        entity  name the french occupation period during the r...
42        entity           what determines the value of the value ?
43        entity         what language is mostly spoken in brazil ?
44        entity  what is the lens behind the iris in the eye ca...
45        entity                    what are halloween ' s colors ?
46        entity     in beetle eaton , can you name sarge ' s dog ?
47        entity  lineman fastener in whitcomb judson patent in ...
48        entity  what game do steve mcqueen and edward g . robi...
49  abbreviation                 what is the full name of the plo ?
50  abbreviation                                      what is dsh ?
51  abbreviation                         what does nasa stand for ?
52  abbreviation                          what does ekg stand for ?
53  abbreviation           what does ` ` o . y . o . b . ' ' mean ?
54  abbreviation  what is the abbreviated name of the national b...
55  abbreviation                          what does bud stand for ?
56  abbreviation  what is p . m . an abbreviation for , as in 10...
57  abbreviation  what is the abbreviation after the company nam.

In [4]:
''' c mod bert (= Bert Only MLM) 결과 '''
cmodbert_dataset_9to10 = pd.read_csv("exp_9_10/cmodbert/cmodbert_aug.tsv", delimiter='\t')
cmodbert_dataset_9to10

numeric  \
0        numeric   
1        numeric   
2        numeric   
3        numeric   
4        numeric   
5        numeric   
6        numeric   
7        numeric   
8        numeric   
9          human   
10         human   
11         human   
12         human   
13         human   
14         human   
15         human   
16         human   
17         human   
18         human   
19      location   
20      location   
21      location   
22      location   
23      location   
24      location   
25      location   
26      location   
27      location   
28      location   
29   description   
30   description   
31   description   
32   description   
33   description   
34   description   
35   description   
36   description   
37   description   
38   description   
39        entity   
40        entity   
41        entity   
42        entity   
43        entity   
44        entity   
45        entity   
46        entity   
47        entity   
48        entity   
49  abbreviation   
50  abbreviation   
51  abbreviation   
52  abbreviation   
53  abbreviation   
54  abbreviation   
55  abbreviation   
56  abbreviation   
57  abbreviation   
58  abbreviation   

   how many games are played in a five - team round -ᅡ tournament ?  
0   what month were you born known if your birthst...                
1   how much will the california be were the year ...                
2         the film ` ` jaws ' ' waszyn in what year ?                
3   how many muscles does the average ensure use w...                
4     when did the chernobyl nuclear accident occur ?                
5   how many degrees cooler is the inside of ` cuc...                
6   is where children under 18 are victims of some...                
7         what year did spielberg make ` ` jaws ' ' ?                
8       how many spears are there on kenya ' l flag ?                
9   whose special bear ' s creator was born on jan...                
10  what major airline has the best safety record ...                
11  what civil war general wreaked havoc on the so...                
12  what is genius ' : ` ` everything comes to him...                
13  in the movie groundshog what day is the name o...                
14  what famous model was married to billy nashvil...                
15  who said ` ` the only thing we have is fear is...                
16          which presidents of the usa were masons ?                
17                who is ` ` the scarlet letter ' ' ?                
18  what is ` ` the airline built for delicately t...                
19  whereutable i find information on the cyclone ...                
20  what country in latin america is the largest o...                
21  where can i find information about cuba barr ,...                
22  what celestial abuse has a diameter of 86 saxo...                
23  what country boasts the 1100 cars per mile of ...                
24  what constellation represents a hunter obe clu...                
25  where is pro [unused389] [unused79] gamble hea...                
26                    what is the capital of kosovo ?                
27                               what is south bend ?                
28  what russian city used to be called stn peters...                
29  how do i e - mail someone at aol . com from ya...                
30  how can i find [unused449] 23rd cities have ca...                
31  how did super president gain his powers on the...                
32                                 how do ihil thee ?                
33  what does the number henry on the rolling 男 be...                
34  what did john f . kennedy consider his amend b...                
35  shooting are peanut butter cookies topped with...                
36                          how illumination lawrence                
37  what was spin tailors , the television show fr...                
38                                what is ariatedul ?                
39      

In [5]:
''' c mod bert p (= Bert MLM + NSM) 결과 '''
cmodbertp_dataset_9to10 = pd.read_csv("exp_0_10/cmodbertp/cmodbertp_aug.tsv", delimiter='\t')
cmodbertp_dataset_9to10

numeric        what is columbia tristar ' s phone number ?
0        numeric                how big is our galaxy in diameter ?
1        numeric  what is the average number of goals scored by ...
2        numeric            how many varieties of apple are there ?
3        numeric             what year did montana become a state ?
4        numeric  how many lawyers are there in the state of new...
5        numeric    what chapter in the bible has the most verses ?
6        numeric             what is dick clark ' s date of birth ?
7        numeric                               what is boxing day ?
8        numeric                        what is the thai new year ?
9          human                 who invented ` ` the muppets ' ' ?
10         human  what how commanders directed the forces during...
11         human  who was the first elected mayor of washington ...
12         human  who played sally rogers on the dick van dyke s...
13         human  what apollo 11 astronaut minded in store while...
14         human  who manufactures the software , ` ` photoshop ...
15         human  who was scrooge ' s dead partner in dickens ' ...
16         human  who was the mother of the man who would not be...
17         human   being awarded the flying fickle finger of fate ?
18         human  what ' s the middle name of movie producer jos...
19      location           where is one ' s corpus callosum found ?
20      location       what small island is northeast of trinidad ?
21      location           what city is located in the city of oz ?
22      location  what ' s the fifth - largest country in the wo...
23      location  where does the par shape of the medieval fish ...
24      location  what in the name of the planet that the ewoks ...
25      location  what asian gulf were the destroyers maddox and...
26      location  what u . s . city is known as the financial ca...
27      location  what country is the setting for edgar allan po...
28      location  what country is cawdor castle , glamis castle ...
29   description                      what is thermal equilibrium ?
30   description  how did the hohezollerns build their power aro...
31   description                         what happened to pompeii ?
32   description                             what is home banking ?
33   description                           what is an urban smirk ?
34   description  what was al capone finally imprisoned for , in...
35   description                          how do birds have wings ?
36   description  what was the v - 8 juice slogan : the ' the ta...
37   description                 why are there olive in martiniss ?
38   description  what is the history of spartacus , the roman s...
39        entity  what kind of patent did ermal arize obtain in ...
40        entity       which ear can most people hear better with ?
41        entity  name the french occupation period during the r...
42        entity           what determines the value of the value ?
43        entity         what language is mostly spoken in brazil ?
44        entity  what is the lens behind the iris in the eye ca...
45        entity                    what are halloween ' s colors ?
46        entity     in beetle eaton , can you name sarge ' s dog ?
47        entity  lineman fastener in whitcomb judson patent in ...
48        entity  what game do steve mcqueen and edward g . robi...
49  abbreviation                 what is the full name of the plo ?
50  abbreviation                                      what is dsh ?
51  abbreviation                         what does nasa stand for ?
52  abbreviation                          what does ekg stand for ?
53  abbreviation           what does ` ` o . y . o . b . ' ' mean ?
54  abbreviation  what is the abbreviated name of the national b...
55  abbreviation                          what does bud stand for ?
56  abbreviation  what is p . m . an abbreviation for , as in 10...
57  abbreviation  what is the abbreviation after the company nam.

---
---
#### * .tsv파일이 그냥 문장으로 구성된 파일인데, 앞에 그 문장의 label? 설명? CLS?이 tab으로 구분되어 저장되어있음
#### * exp_0_10, exp_1_10, ..., exp_14_10은 전체 trec 데이터셋을 총 15번 나누어 저장한거고, 
#### * 각 label? CLS? 별로 10개씩 끊어서 했다는 것임.


---
---
##### L2 + L3 데이터셋을 그냥 쓸 수 있는지 체크하려면
1. 일단 그대로 넣어서 돌려볼것
    -> 단, scripts/bert_trec_lower.sh와 cmodbert.py와 cmodbertp.py 를 "새로운 파일로 복사!!"해와서 
    -> "데이터셋명"이랑 "경로" 싹 바꿔줄것!!!!!!!!!!
-
-
2. cmodbert.py와 cmodbertp.py 파일에서 dataset을 [Tab]으로 구분하는 부분이 있는지 찾아볼것
    -> 그걸 주석처리한다면 L2를 그대로 사용해도 되지 않을까???
-
-
3. 안된다면 L2 + L3 데이터 안에 numeric, human, description 등을 각 문장에 추가하고 [Tab]으로 구분해서 만들고 돌려볼것



### 지금 한거
- Dataset에 L2폴더 만들고 실행 .sh파일 task를 L2로 바꿔줌(그냥 디렉터리만 바꿔주는 것 같음)

- 그안에 exp_0_10, exp_1_10, exp_2_10, .. 를 각각 만들어줌
 -> 이 안에 train.tsv, test.tsv, dev.tsv 각각 어떻게 만들어줄지 모르겠음

- 이렇게만 데이터구성해서 돌리면 수행될까

---
---

train.tsv는 원래 총 4905개 문장임

exp_0_10/train.tsv 와 각 bert_augment.tsv는
  6개 클래스 각10개씩 => exp_i_j는 60개의 문장씩 들어감

